src/extension.ts

```diff
@@ -183,3 +84,102 @@ function getDashOption(): DashOption {
     exactDocset,
   };
 }
+
+/**
+ * Get docset configuration
+ *
+ * @param {string} languageId e.g javascript, ruby
+ * @return {string}
+ */
+function getDocsets(): string[] {
+  const editor = getEditor();
+  const fileName = path.basename(editor!.document.fileName);
+  const languageId = editor!.document.languageId;
+
+  const fileNameDocsets = getFileNameDocsets(fileName);
+  const languageIdDocsets = getLanguageIdDocsets(languageId);
+
+  // prioritize docset matching by file name then language id
+  return [...fileNameDocsets, ...languageIdDocsets] as string[];

```

Comment:
is `as string[]` necessary?

Reply:
i know what you mean. that looks unnecessary for me too but for some reason TS identifies that as `(string | undefined)[]`

Reply:
it's all good now. i think it was hiccup